relay-event-normalization/src/normalize/user_agent.rs

```diff
@@ -46,10 +46,48 @@ pub fn normalize_user_agent_info_generic(
     platform: &Annotated<String>,
     user_agent_info: &RawUserAgentInfo<&str>,
 ) {
-    if !contexts.contains::<BrowserContext>() {
+    // Check if a browser context already exists
+    let existing_browser_context = contexts.contains::<BrowserContext>();
+
+    if !existing_browser_context {
         if let Some(browser_context) = BrowserContext::from_hints_or_ua(user_agent_info) {
             contexts.add(browser_context);
         }
+    } else if let Some(browser_context) = contexts.get_mut::<BrowserContext>() {
```

Comment:
This already tells you if you have a context or not, meaning you don't need the contains above.